examples/App.re

```diff
@@ -9,69 +7,79 @@ type state = {route};
 type action =
   | UpdateRoute(route);
 
-let getInitialRoute = () =>
-  switch (locationHash) {
-  | "#signup" => Signup
-  | "#login" => Login
-  | _ => Signup
-  };
+[@react.component]
+let make = () => {
+  let initialRoute =
+    React.useMemo1(
+      () => {
+        let url = ReasonReactRouter.dangerouslyGetInitialUrl();
+        switch (url.hash) {
+        | "signup" => Signup
+        | "login" => Login
+        | _ => Signup
+        };
+      },
+      [||],
+    );
 
-let component = React.reducerComponent(__MODULE__);
+  let (state, dispatch) =
+    React.useReducer(
+      (_, action) =>
+        switch (action) {
+        | UpdateRoute(nextRoute) => {route: nextRoute}
+        },
+      {route: initialRoute},
+    );
 
-let make = _ => {
-  ...component,
-  initialState: () => {route: getInitialRoute()},
-  reducer: (action, _) =>
-    switch (action) {
-    | UpdateRoute(route) => React.Update({route: route})
+  React.useEffect1(
+    () => {
+      let watcherID =
+        ReasonReactRouter.watchUrl(url =>
```

Comment:
Why not `ReasonReactRouter.useUrl`? There is a small issue with this implementation in concurrent mode which is that this effect is async. If the url changes between mount and the effect you won't detect it. `useUrl` accounts for this case - and also removes a lot of boilerplate :)

Reply:
TIL as well, will use it 👍